### PR TITLE
add useFollowProfile and useUnfollowProfile hooks

### PR DESCRIPTION
**Pull request by krzysu**
follow and unfollow as a signed txs only but they work!

### CODE REVIEW

File: packages/react/src/transactions/adapters/useUnfollowProfileController.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  > => {
+    validateUnfollowRequest(request);
+
+    const presenter = new TransactionResultPresenter<
+      UnfollowRequest,
+      BroadcastingError | PendingSigningRequestError | UserRejectedError | WalletConnectionError
+    >();
+    const gateway = new UnfollowProfileCallGateway(apolloClient);
+
+    const unfollowProfiles = new UnfollowProfile(
```
Comment (cesarenaldi) [inline packages/react/src/transactions/adapters/useUnfollowProfileController.ts:38]: nit: 
```suggestion
    const unfollowProfile = new UnfollowProfile(
```

File: packages/react/src/transactions/adapters/useFollowProfileController.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+      onChainRelayer,
+      transactionQueue,
+      presenter,
+    );
+
+    const balanceGateway = new BalanceGateway(providerFactory);
+    const tokenGateway = new TokenGateway(providerFactory);
+    const tokenAvailability = new TokenAvailability(balanceGateway, tokenGateway, activeWallet);
+
+    const followProfiles = new FollowProfile(tokenAvailability, signedFollow, presenter);
```
Comment (cesarenaldi) [inline packages/react/src/transactions/adapters/useFollowProfileController.ts:72]: nit: 
```suggestion
    const followProfile = new FollowProfile(tokenAvailability, signedFollow, presenter);
```

File: packages/react/src/transactions/useFollowProfile.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+  // );
+
+  return useDeferredTask(async ({ profile }) => {
+    // if (hasPendingUnfollowTx) {
+    //   return failure(
+    //     new PrematureFollowError(
+    //       `A previous unfollow request for ${profile.id} is still pending. Make sure you check 'profile.operations.canFollow' beforehand.`,
+    //     ),
+    //   );
+    // }
```
Comment (cesarenaldi) [inline packages/react/src/transactions/useFollowProfile.ts:98]: not blocking: what's the plan for `PrematureFollowError`?

File: packages/react/src/profile/useProfileManagers.ts [hunk 4]
```diff
@@ -39,6 +39,9 @@ export type UseProfileManagersArgs = PaginatedArgs<{
  *   );
  * }
  * ```
+ *
+ * @category Profiles
+ * @group Hooks
```
Comment (cesarenaldi) [inline packages/react/src/profile/useProfileManagers.ts:44]: 🙏

File: packages/react/src/transactions/useUpdateFollowPolicy.ts [hunk 5]
```diff
@@ -20,6 +20,47 @@ export type UpdateFollowPolicyArgs = {
...
+ *
+ * await execute({
+ *   followPolicy: {
+ *     type: FollowPolicyType.NO_ONE,
+ *   },
+ * });
+ * ```
+ *
+ * @example
+ * Anyone can follow, but they must pay a fee.
```
Comment (cesarenaldi) [inline packages/react/src/transactions/useUpdateFollowPolicy.ts:51]: Can you please add a little explainer about `Amount` with link like in here: https://github.com/lens-protocol/lens-sdk/blob/lens-v2/packages/react/src/transactions/useCreatePost.ts#L170-L183

Copy and paste is fine. Better over communicate at this point.

And also clarify the example lile in `useCreatePost` (see link above).

File: packages/react/src/transactions/adapters/responders/FollowProfileResponder.tsx [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+          ...current.operations,
+          canFollow: TriStateValue.No,
+          isFollowedByMe: {
+            __typename: 'OptimisticStatusResult',
+            value: true,
+            isFinalisedOnchain: false,
+          },
+        },
+      };
+    });
```
Comment (cesarenaldi) [inline packages/react/src/transactions/adapters/responders/FollowProfileResponder.tsx:31]: These should be overidden via field policies to achieve a reliable result on refresh or other tabs.

File: packages/react/src/transactions/adapters/responders/UnfollowProfileResponder.ts [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+          ...current.operations,
+          canUnfollow: false,
+          isFollowedByMe: {
+            __typename: 'OptimisticStatusResult',
+            value: false,
+            isFinalisedOnchain: false,
+          },
+        },
+      };
+    });
```
Comment (cesarenaldi) [inline packages/react/src/transactions/adapters/responders/UnfollowProfileResponder.ts:30]: Same here.... field policies are more reliable on refreshes and other threads.

**Comment by krzysu** [comment]
@cesarenaldi comments from this PR I'm addressing in the follow-up one https://github.com/lens-protocol/lens-sdk/pull/610

**Comment by krzysu** [comment]
new field policies are in this PR https://github.com/lens-protocol/lens-sdk/pull/611

File: packages/react/src/transactions/useUpdateFollowPolicy.ts (hunk 5 above)
Comment (krzysu) [reply packages/react/src/transactions/useUpdateFollowPolicy.ts:51]: done in https://github.com/lens-protocol/lens-sdk/pull/610/commits/daf67ba715df13142221e2de80356d54edf4bb91

File: packages/react/src/transactions/adapters/responders/FollowProfileResponder.tsx (hunk 6 above)
Comment (krzysu) [reply packages/react/src/transactions/adapters/responders/FollowProfileResponder.tsx:31]: see https://github.com/lens-protocol/lens-sdk/pull/611

File: packages/react/src/transactions/useFollowProfile.ts (hunk 3 above)
Comment (krzysu) [reply packages/react/src/transactions/useFollowProfile.ts:98]: how to resolve this?

Screenshot 2023-10-17 at 17 59 16

**Pull request merged**